### ISSUE TITLE
Find blueprint when returning alot of results crashes sails

### ISSUE THREAD

**Issue by alexjp** [balderdashy/sails#2621]
Hi,

I am testing sails with the blueprints.
I have a model User ( pretty standard attributes and some associations ) that i made 3000 results.
If limit option is set to more than 3000 results, doing the find request, sails crashs:

`
/home/alexandre/Actividades/novo_assistencias/project/server/node_modules/sails-mysql/node_modules/mysql/lib/protocol/Parser.js:82
        throw err;
              ^
RangeError: Maximum call stack size exceeded
    at /home/alexandre/Actividades/novo_assistencias/project/server/node_modules/sails-mysql/lib/adapter.js:612:67
    at Array.forEach (native)
    at splitChildren (/home/alexandre/Actividades/novo_assistencias/project/server/node_modules/sails-mysql/lib/adapter.js:610:37)
    at iterate (/home/alexandre/Actividades/novo_assistencias/project/server/node_modules/sails-mysql/node_modules/async/lib/async.js:149:13)
    at /home/alexandre/Actividades/novo_assistencias/project/server/node_modules/sails-mysql/node_modules/async/lib/async.js:160:25
    at splitChildren (/home/alexandre/Actividades/novo_assistencias/project/server/node_modules/sails-mysql/lib/adapter.js:616:21)
    at iterate (/home/alexandre/Actividades/novo_assistencias/project/server/node_modules/sails-mysql/node_modules/async/lib/async.js:149:13)
    at /home/alexandre/Actividades/novo_assistencias/project/server/node_modules/sails-mysql/node_modules/async/lib/async.js:160:25
    at splitChildren (/home/alexandre/Actividades/novo_assistencias/project/server/node_modules/sails-mysql/lib/adapter.js:616:21)
    at iterate (/home/alexandre/Actividades/novo_assistencias/project/server/node_modules/sails-mysql/node_modules/async/lib/async.js:149:13)
    at /home/alexandre/Actividades/novo_assistencias/project/server/node_modules/sails-mysql/node_modules/async/lib/async.js:160:25
    at splitChildren (/home/alexandre/Actividades/novo_assistencias/project/server/node_modules/sails-mysql/lib/adapter.js:616:21)
    at iterate (/home/alexandre/Actividades/novo_assistencias/project/server/node_modules/sails-mysql/node_modules/async/lib/async.js:149:13)
    at /home/alexandre/Actividades/novo_assistencias/project/server/node_modules/sails-mysql/node_modules/async/lib/async.js:160:25
    at splitChildren (/home/alexandre/Actividades/novo_assistencias/project/server/node_modules/sails-mysql/lib/adapter.js:616:21)
    at iterate (/home/alexandre/Actividades/novo_assistencias/project/server/node_modules/sails-mysql/node_modules/async/lib/async.js:149:13)
`

Removing the 
`query = actionUtil.populateEach(query, req);`
makes it work again.

**Comment by sgress454** [comment]
Thanks @alexjp.  Can you post the models you're seeing this with (or better yet, just a link to your code if it's public) so we can run some tests? cc: @particlebanana

**Comment by alexjp** [comment]
Hi,

I cannot put the code public, but i can put the involved models for sure:

https://gist.github.com/alexjp/86d08b30390895aec726

hope it helps!

**Comment by CWyrtzen** [comment]
We'd like to take a closer look at this but can you give me an update on the current status, any resolutions or workarounds, and how this issue behaves after upgrade? Thanks~

**Comment by alexjp** [comment]
Hi,

The workaround i used at the time was to only allow results to return 100 or less items. ( especially if relationships were involved ).

Unfortunatly at this time i am not working in that application anymore. If this problem is not happening to anyone else, i can create a new application and test it.

**Comment by sailsbot** [comment]
Thanks for posting, @alexjp.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
